core/src/server/rpc_module.rs

```diff
@@ -379,107 +411,128 @@ impl Methods {
 
 	/// Make a request (JSON-RPC method call or subscription) by using raw JSON.
 	///
-	/// Returns the raw JSON response to the call and a stream to receive notifications if the call was a subscription.
+	/// This is a low-level API where you might be able build "in-memory" client
+	/// but it's not ergonomic to use, see [`Methods::call`] and [`Methods::subscribe_bounded`]
+	/// for more ergonomic APIs to perform method calls or subscriptions for testing or other purposes.
+	///
+	/// After this method returns the responses can be read on the corresponding `Receiver` to the `Sender`
+	/// that was passed into to the call.
 	///
 	/// # Examples
 	///
 	/// ```
 	/// #[tokio::main]
 	/// async fn main() {
-	///     use jsonrpsee::RpcModule;
+	///     use jsonrpsee::{RpcModule, SubscriptionMessage};
 	///     use jsonrpsee::types::Response;
 	///     use futures_util::StreamExt;
 	///
 	///     let mut module = RpcModule::new(());
-	///     module.register_subscription("hi", "hi", "goodbye", |_, mut sink, _| {
-	///         sink.send(&"one answer").unwrap();
-	///         Ok(())
+	///
+	///     module.register_subscription("hi", "hi", "goodbye", |_, pending, _| async move {
+	///          let sink = pending.accept().await?;
+	///          let msg = SubscriptionMessage::from_json(&"one answer").unwrap();
+	///          sink.send(msg).await.unwrap();
+	///
+	///          Ok(())
 	///     }).unwrap();
-	///     let (resp, mut stream) = module.raw_json_request(r#"{"jsonrpc":"2.0","method":"hi","id":0}"#).await.unwrap();
-	///     let resp = serde_json::from_str::<Response<u64>>(&resp.result).unwrap();
-	///     let sub_resp = stream.next().await.unwrap();
+	///
+	///     let (tx, mut rx) = tokio::sync::mpsc::channel(16);
+	///     module.raw_json_request(r#"{"jsonrpc":"2.0","method":"hi","id":0}"#, tx).await.unwrap();
+	///     let sub_resp = rx.recv().await.unwrap();
+	///     // the subscription ID is sent in the `result` field.
+	///     let sub_resp = serde_json::from_str::<Response<u64>>(&sub_resp).unwrap();
+	///
+	///     let sub_notif = rx.recv().await.unwrap();
 	///     assert_eq!(
-	///         format!(r#"{{"jsonrpc":"2.0","method":"hi","params":{{"subscription":{},"result":"one answer"}}}}"#, resp.result),
-	///         sub_resp
+	///         format!(r#"{{"jsonrpc":"2.0","method":"hi","params":{{"subscription":{},"result":"one answer"}}}}"#, sub_resp.result),
+	///         sub_notif
 	///     );
 	/// }
 	/// ```
-	pub async fn raw_json_request(
-		&self,
-		request: &str,
-	) -> Result<(MethodResponse, mpsc::UnboundedReceiver<String>), Error> {
+	pub async fn raw_json_request(&self, request: &str, tx: mpsc::Sender<String>) -> Result<(), Error> {
```

Comment:
changed this API, have a look

Reply:
I do prefer this to the old API; no harder to use and gives more control to the caller over the boundedness :)